mdsuite/calculators/calculator.py

```diff
@@ -364,10 +378,26 @@ def _msd_operation(ensemble: tf.Tensor, square: bool = True):
                 Mean square displacement.
         """
         if square:
-            return tf.math.squared_difference(ensemble, ensemble[:, None, 0])
+            return tf.math.squared_difference(np.array(ensemble)[:, self.tau_values], ensemble[:, None, 0])
         else:
             return tf.math.subtract(ensemble, ensemble[:, None, 0])
 
+    def _handle_tau_values(self):
+        """
+        Handle the parsing of custom tau values.
+
+        Returns
+        -------
+        Updates the class state.
+        """
+        if type(self.tau_values) is int:
+            self.data_resolution = self.tau_values
+            self.tau_values = np.linspace(0, self.data_range - 1, int(self.tau_values), dtype=int)
+        if type(self.tau_values) is list:
+            self.data_resolution = len(self.tau_values)
+        if type(self.tau_values) is slice:
+            self.data_resolution = len(np.linspace(0, self.data_range, self.data_range)[self.tau_values])
```

Comment:
Isn't this just `len(np.arange(self.data_range)[self.tau_values])` 
I don't know the type slice, but I guess `len(self.tau_values)` does not work?

Reply:
no it's okay how it is. The data resolution is just a scaling factor to re-scale the time arrays or msd arrays. 

```python
time = np.linspace(0, self.data-range, self.data-resolution)
```

Reply:
Okay, I checked and

```py
 np.linspace(0, x, x - 1) == np.arange(x)
```

So I guess this is resolved then